### PR TITLE
[INFRANG-5407]: wag: a model that's just "type:string" generates inva…

### PR DESCRIPTION
**Pull request by philippeclever**
…lid code
# JIRA
https://clever.atlassian.net/browse/INFRANG-5407

# Description
Update the definition to use new(*model.Type) instead of modes.Type{}

# Testing
unit tests added to improve test coverage

#

Todo:

- [x] Run `make build`
- [x] Run `make generate`
- [ ] Update the current version in the `/VERSION` file.

### CODE REVIEW

**Comment by taylor-sutton** [comment]
> unit tests added to improve test coverage

I don't see any tests, did you mean to add new files?

File: samples/Makefile [hunk 1]
```diff
@@ -23,6 +23,9 @@ generate:
 	../bin/wag -file ./nils.yml -output-path ./gen-go-nils -js-path ./gen-js-nils
 	cd ./gen-js-nils && jsdoc2md index.js types.js > ./README.md
 	go generate -mod=mod ./gen-go-nils...
+	../bin/wag -file ./strings.yml -output-path ./gen-go-strings -js-path ./gen-js-strings
```
Comment (taylor-sutton) [inline samples/Makefile:26]: This `.yml` doesn't match the filename `.yaml`

**Comment by philippeclever** [comment]
@taylor-sutton test cases has been added and validated.

**Review by taylor-sutton — APPROVED** [review]
LGTM!

**Pull request merged**